alt-tab-macos/ui/PreferencesPanel.swift

```diff
@@ -1,110 +1,211 @@
 import Cocoa
 
-class PreferencesPanel: NSPanel, NSTextViewDelegate {
-    var maxScreenUsage: NSTextView?
-    var windowPadding: NSTextView?
-    var cellPadding: NSTextView?
-    var cellBorderWidth: NSTextView?
-    var maxThumbnailsPerRow: NSTextView?
-    var iconSize: NSTextView?
-    var fontHeight: NSTextView?
-    var interItemPadding: NSTextView?
-    var tabKeyCode: NSTextView?
-    var windowDisplayDelay: NSTextView?
+class PreferencesPanel: NSPanel {
+    // ui: base layout
+    let panelWidth = CGFloat(400)
+    let panelHeight = CGFloat(400) // gets auto adjusted to content height
+    let panelPadding = CGFloat(40)
+    let panelWidthToLabelRatio = CGFloat(0.5)
+
+    // ui: preferences elements
+    var maxScreenUsage: NSSlider?
+    var maxThumbnailsPerRow: NSTextField?
+    var iconSize: NSSlider?
+    var fontHeight: NSSlider?
+    var tabKeyCode: NSTextField?
+    var windowDisplayDelay: NSTextField?
     var metaKey: NSPopUpButton?
     var theme: NSPopUpButton?
     var showOnScreen: NSPopUpButton?
-    var inputsMap = [NSTextView: String]()
+
+    var invisibleTextField: NSTextField? // default firstResponder and used for triggering of focus loose
 
     override init(contentRect: NSRect, styleMask style: StyleMask, backing backingStoreType: BackingStoreType, defer flag: Bool) {
-        super.init(contentRect: contentRect, styleMask: style, backing: backingStoreType, defer: flag)
+        let initialRect = NSRect(x: 0, y: 0, width: panelWidth, height: panelHeight)
+        super.init(contentRect: initialRect, styleMask: style, backing: backingStoreType, defer: flag)
         title = Application.name + " Preferences"
+        titlebarAppearsTransparent = true
         hidesOnDeactivate = false
-        contentView = makeGridView(makeLabelsAndInputs(), makeWarningLabel())
+        contentView = makeContentView()
+
+        // setup hidden element
+        invisibleTextField = NSTextField()
+        invisibleTextField?.isHidden = true
+        contentView?.subviews.append(invisibleTextField!)
+        self.initialFirstResponder = invisibleTextField
```

Comment:
The inline comment is a tell that this would be made clearer be being in a sub-method, the same way right above there is a `makeContentView` method and not a block of instructions with an inline comment `// make content view` ;p

Reply:
Gotcha. Made it non atomic, so it executes this four commands.